[BUG] callProcedure does not work on Windows.

**Describe your system**
* `odbc` Package Version: 2.3.4
* ODBC Driver: IBM i Access ODBC Driver (13.64.24)
* Database Name: Db2 for i
* Database Version: x
* Database OS: IBM i
* Node.js Version: 12.13.1
* Node.js OS: Windows

**Describe the bug**

I am getting, what could be an encoding issue with odbc.

```
[IBM][System i Access ODBC Driver][DB2 for i5/OS]SQL0104 - Token  was not valid. Valid tokens: ( CL END GET SET TAG CALL DROP FREE HOLD LOCK OPEN WITH ALTER.
````

When running: `const results = await connection.callProcedure(null, 'BARRY', 'ABCD_PGM', [ds, '']);`

But I can run the usual `executeStatement` with no problems.

I am only getting this issue on Windows. I tested on my Macbook and the same code works.

**Expected behavior**
Return of output parameters.

![image](https://user-images.githubusercontent.com/3708366/91585105-83b94e80-e94b-11ea-88cf-8cbd5c3924a2.png)

**To Reproduce**
Steps to reproduce the behavior:
1. Connect to database with options:

```js
  connectionString = [
    `Driver=IBM i Access ODBC Driver`,
    `System=${process.env.DB2_SYSTEM}`,
    `UID=${process.env.DB2_UID}`,
    `Password=${process.env.DB2_PASSWORD}`,
    `DBQ=,*USRLIBL`,
    `Naming=1`,
  ].join(';'),
```

2. Call a stored procedure with one in parameter and one out parameter. (as above)

**Additional context**

>The source looked good, but everything got converted to 3f...

```
t=e89c 15:19:39.373 [00]                  NLS: NL CONV:cvt=6107 scp=1202 tcp=273 siso=1 pad=4 s*=0x205709fc9a0 sl=18 t*=0x20570a12740 tl=30
t=e89c 15:19:39.373 [00]                  NLS: src:18 
[00001] 7b2043414c4c20422e4120283f2c3f29207d                              <{ CALL B.A (?,?) }              > [#...<<...........'              ]
t=e89c 15:19:39.373 [00]                  NLS: tgt:9 
[00001] 3f3f3f3f3f3f3f3f3f                                                <?????????                       > [.........                       ]
```

> The text comes back in EBCDIC, the important hex is here: 403f40. space, substitution character, space.

```
[00002] f1f0f40000006d380201110063e396928595403f40a681a2409596a340a58193  <......m8....c.....@?@...@...@...> [104..._......Token . was not val]
```

![image](https://user-images.githubusercontent.com/3708366/91585442-f9251f00-e94b-11ea-8297-514b8a2d8f0a.png)

The hex of the data is "7b2043414c4c20422e4120283f2c3f29207d" but from the trace it says that it's converting from CCSID 1202, which is UTF-16LE, but that hex is **absolutely not** UTF-16.

The problem is the SQL is being generated [using sprintf](https://github.com/markdirish/node-odbc/blob/master/src/odbc_connection.cpp#L1812-L1813), which creates a standard C string, while the API is expecting a wide string.

I think you'd need something like
```C
#ifdef UNICODE
swprintf
swprintf(data->sql, 255, L"{ CALL %S (%S) }", combinedProcedureName, parameterString);
#else
sprintf((char *)data->sql, "{ CALL %s (%s) }", combinedProcedureName, parameterString);
#endif
```

There's some other issues mentioned in the build logs:
```
Warning: D:\a\node-odbc\node-odbc\src\odbc_connection.cpp(1828,34): warning C4477: 'sprintf' : format string '%s' requires an argument of type 'char *', but variadic argument 2 has type 'SQLTCHAR *' [D:\a\node-odbc\node-odbc\build\odbc.vcxproj]
D:\a\node-odbc\node-odbc\src\odbc_connection.cpp(1828,34): message : consider using '%ls' in the format string [D:\a\node-odbc\node-odbc\build\odbc.vcxproj]
D:\a\node-odbc\node-odbc\src\odbc_connection.cpp(1828,34): message : consider using '%lls' in the format string [D:\a\node-odbc\node-odbc\build\odbc.vcxproj]
D:\a\node-odbc\node-odbc\src\odbc_connection.cpp(1828,34): message : consider using '%Ls' in the format string [D:\a\node-odbc\node-odbc\build\odbc.vcxproj]
D:\a\node-odbc\node-odbc\src\odbc_connection.cpp(1828,34): message : consider using '%ws' in the format string [D:\a\node-odbc\node-odbc\build\odbc.vcxproj]
```

--- a/src/odbc_connection.cpp
+++ b/src/odbc_connection.cpp
@@ -1797,20 +1797,27 @@ class CallProcedureAsyncWorker : public ODBCAsyncWorker {
 
       // create the statement to call the stored procedure using the ODBC Call escape sequence:
       // need to create the string "?,?,?,?" where the number of '?' is the number of parameters;
-      SQLTCHAR *parameterString = new SQLTCHAR[255]();
+      char *parameterString = new char[255];
 
       for (int i = 0; i < data->parameterCount; i++) {
         if (i == (data->parameterCount - 1)) {
-          strcat((char *)parameterString, "?"); // for last parameter, don't add ','
+          strcat(parameterString, "?"); // for last parameter, don't add ','
         } else {
-          strcat((char *)parameterString, "?,");
+          strcat(parameterString, "?,");
         }
       }
 
       data->deleteColumns(); // delete data in columns for next result set
 
-      data->sql = new SQLTCHAR[255]();
+      data->sql = new SQLTCHAR[255];
+#ifndef UNICODE
       sprintf((char *)data->sql, "{ CALL %s (%s) }", combinedProcedureName, parameterString);
+#else
+      // Note: On Windows, %s and %S change their behavior depending on whether
+      // it's passed to a printf function or a wprintf function. Since we're passing
+      // narrow strings to a wide function, we need to use %S.
+      swprintf(data->sql, 255, L"{ CALL %S (%S) }", combinedProcedureName, parameterString);
+#endif
 
       delete[] combinedProcedureName;
       delete[] parameterString;

